src/components/token-rewards/AddTimeframeStep.tsx

```diff
@@ -0,0 +1,126 @@
+        <div className="flex flex-row items-center justify-between gap-3 mb-2">
+          <h1 className="text-lg font-semibold text-text-lightmode-primary dark:text-text-darkmode-primary">
+            Add Timeframe
+          </h1>
+          {isMobile && (
+            <div className="flex flex-row items-center gap-3">
+              <Icon
+                src="/img/assets/question-icn.svg"
+                alt="help"
+                className="w-[30px] h-[30px] cursor-pointer"
+                onClick={() => window.open('https://www.goosefx.io/gamma#faqs')}
+              />
+              <DialogClose>
+                <Icon src="/img/assets/rewards_close.svg" alt="Close" className="w-4 h-4" />
+              </DialogClose>
+            </div>
+          )}
+        </div>
+
+        <p className="text-sm text-text-lightmode-secondary dark:text-text-darkmode-secondary mb-6">
+          Choose a date and time limit for token rewards, all timeframes are based on UTC time.
+        </p>
+
+        {connected ? (
+          <>
+            {selectedToken ? (
+              <div className="grid grid-cols-[auto_auto_auto] w-max gap-1 items-center">
+                <div className="flex flex-row items-center gap-3 mb-2">
+                  <IconWithFallback
+                    src={loadIconImage(selectedToken?.logoURI, mode)}
+                    size={'sm'}
+                    className={'rounded-circle'}
+                  />
+                  <p>{selectedToken?.symbol}</p>
+                </div>
+                <div className="col-span-2">
+                  <p className="text-end"> {dayjs(endDate).diff(startDate, 'days')} Days</p>
+                </div>
+                <DateTimeInputWithDialog value={startDate} onChange={setStartDate} />
+                <p>to</p>
+                <DateTimeInputWithDialog value={endDate} onChange={setEndDate} />
+              </div>
+            ) : (
+              <Button
+                onClick={() => setCurrentStep(currentStep - 1)}
+                className="px-4 py-2 text-text-lightmode-primary dark:text-text-darkmode-primary"
+              >
+                Back
+              </Button>
+            )}
+          </>
+        ) : (
+          <Connect containerStyle="w-max" />
+        )}
+
+        <div className="flex justify-between pt-8 mt-auto">
+          <Button
+            onClick={() => setCurrentStep(currentStep - 1)}
+            className="px-4 py-2 text-text-lightmode-primary dark:text-text-darkmode-primary"
+          >
+            Back
+          </Button>
+          {startDate && endDate && isMobile && (
+            <Button
+              className="px-4 py-2 cursor-pointer"
+              colorScheme={'blue'}
+              variant={'secondary'}
+              onClick={() => setCurrentStep(currentStep + 1)}
+            >
+              Next
+            </Button>
+          )}
+        </div>
+      </div>
+
+      {!isMobile && (
```

Comment:
Should this be isMobile? As Dialog is mobile if I recall correctly :)

Reply:
just commenting again @NishantChandla if this should be just be `isMobile` not `!isMobile` as it references DialogClose which is only rendered on Mobile

Reply:
What I mean is on line 146 we are doing !isMobile and rendering a dialog close; but above on line 53 we are checking for isMobile. I just want to confirm that the bottom check is correct unless I am missing something with line 146 or 53 :)

Reply:
Awesome just wanted to confirm as the bool was flipped but did the same stuff :)